usrp_drivers/n200/n200_driver.cpp

```diff
@@ -498,28 +441,28 @@ void receive(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &driver
   auto receive_channels = driver_options.get_receive_channels();
   stream_args.channels = receive_channels;
 
-  usrp_d.set_rx_rate(rx_rate_hz, receive_channels);  // ~450us
+
   uhd::rx_streamer::sptr rx_stream = usrp_d.get_usrp_rx_stream(stream_args);  // ~44ms
 
-  /* 100 is the arbitrary scaling for the usrp_buffer_size
-     so there won't be fragmentation and the while(1) loop below
-     with the recv runs less times
-  */
-  auto usrp_buffer_size = 100 * rx_stream->get_max_num_samps();
-  // TODO: Put ringbuffer_size into the config file
+  auto usrp_buffer_size = rx_stream->get_max_num_samps();
   /* The ringbuffer_size is calculated this way because it's first truncated (size_t)
      then rescaled by usrp_buffer_size */
-  size_t ringbuffer_size = (size_t(500.0e6)/sizeof(std::complex<float>)/usrp_buffer_size) *
-                            usrp_buffer_size;
+  size_t ringbuffer_size = size_t(driver_options.get_ringbuffer_size()/
+                            sizeof(std::complex<float>)/
+                            usrp_buffer_size) * usrp_buffer_size;
+
+  SharedMemoryHandler shrmem(driver_options.get_ringbuffer_name());
+
+  shrmem.create_shr_mem(receive_channels.size() * ringbuffer_size * sizeof(std::complex<float>));
 
-  buffer.resize(receive_channels.size() * ringbuffer_size);
 
   std::vector<std::complex<float>*> buffer_ptrs_start;
 
   for(uint32_t i=0; i<receive_channels.size(); i++){
-    auto ptr = static_cast<std::complex<float>*>(buffer.data() + (i * ringbuffer_size));
+    auto ptr = static_cast<std::complex<float>*>(shrmem.get_shrmem_addr()) + (i * ringbuffer_size);
```

Comment:
There's another location where you calculate a pointer where the parenthesis is in the wrong place, I can't remember where, but it's in this commit and I forgot to comment on it.

Reply:
Okay I thought I updated all this. I think this only had to change since the shared memory addr is a void*